### DISCUSSION TITLE
Ore/Resource pathfinding

### DISCUSSION THREAD

**Discussion by Splarkszter** [ORelio/Minecraft-Console-Client#2178] [Q&A]
Is there a way to detect Ores and pathfind to them and mine them automatically?, i saw a video from a guy that uses something using the command #mine and it goes to the specified ore, very fancy, but seems it uses it from the client so i guess it can't be used in the console client?

Also that it kills mobs in the process since it will be alone doing it's thing.

**Comment by milutinke** [comment]
There is a bot that can mine ores: https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/OreMiner.cs
Although it's not advanced, you need to specify the coordinates of ores.
As for the killing mobs, you have `AutoAttack` bot.

We have plans to port baritone, so in the future there could be a bot that could find and mine ores like Baritone.
Currently it's possible, but it's a bit of a complex thing to make, you can use the `GetWorld().FindBlock` method to find ores.
The bot would need a state machine for any complex decision making, would need to be pretty close to ores, aka on the bottom of the world.
